packages/x-components/src/__stubs__/filters-stubs.factory.ts

```diff
@@ -375,3 +375,49 @@ export function createNextEditableNumberRangeFilter(
     selected: range.min !== null || range.max !== null
   };
 }
+
+/**
+ * Type of a function that creates
+ * {@link @empathyco/x-types-next#HierarchicalFilter | HierarchicalFilter}s. Based on its label,
+ * selected value and children.
+ */
+export type CreateNextHierarchicalFilter = (
+  label: string,
+  selected: boolean,
+  createChildren?: (createChildren: CreateNextHierarchicalFilter) => NextHierarchicalFilter[]
+) => NextHierarchicalFilter[];
+
+/**
+ * Creates a factory of
+ * {@link @empathyco/x-types-next#HierarchicalFilter | HierarchicalFilter}s.
+ *
+ * @param facetId - Facet id to which the filter belongs.
+ * @param parentId - Filter's parent id if exists.
+ * @returns A scoped function which is able to create `HierarchicalFilters`.
+ */
+export function createNextHierarchicalFilterFactory(
+  facetId: string,
+  parentId: string | null = null
+): CreateNextHierarchicalFilter {
+  return (label, selected, createChildren) => {
```

Comment:
If you had added the type, then you would realize about the `callbackInfo` and `value` fields
```suggestion
  return (label, selected, createChildren): NextHierarchicalFilter[] => {
```

Reply:
I didn't know it was using relaxed types when it inferred return types. Good catch